crates/control/src/localization.rs

```diff
@@ -177,22 +185,26 @@ impl Localization {
                 Some(GamePhase::PenaltyShootout {
                     kicking_team: Team::Opponent,
                 }),
-            ) => {
-                let penalty_shoot_out_keeper_pose =
-                    Pose2::from(point![-context.field_dimensions.length / 2.0, 0.0]);
-                self.hypotheses = vec![ScoredPose::from_isometry(
-                    penalty_shoot_out_keeper_pose,
-                    *context.initial_hypothesis_covariance,
-                    *context.initial_hypothesis_score,
-                )];
-                self.hypotheses_when_entered_playing
-                    .clone_from(&self.hypotheses);
+                _,
+            )
+            | (_, _, _, Some(SubState::PenaltyKick)) => {
+                if matches!(kicking_team, Some(Team::Opponent)) {
+                    let penalty_shoot_out_keeper_pose =
+                        Pose2::from(point![-context.field_dimensions.length / 2.0, 0.0]);
```

Comment:
This doesn't work because the keeper is not placed in the middle of the goal. The keeper is only placed backward onto the goal line without being moved in the Y axis.
The only exception is if the goalkeeper is in front of the goalpost at which point it is placed on the goal line on the side of the goalpost that is closer, which may be an idea for a next iteration

Reply:
Now only modifies the x component and matches only for `PlayerNumber::One`